research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,153 @@
+
+## Visual examples
+
+> Provide, if possible, visual examples of the component being built, to provide an idea of implementations.
+
+## DOM hierarchy/structure
+
+> Define the recommended DOM shape to represent the component, and how important attributes (aria attributes especially) are applied to the various parts. In cases where a component nests other components, ideally, you expand the full dom structure to understand both the expectation and if there are shortcommings in the child components' customizability.
+>
+> Example:
+>
+> ```html
+> <div class="root">
+>   <div class="rail"></div>
+>
+>   <div class="mark">
+>     <label class="markLabel" />
+>   </div>
+>   <div class="track" />
+>   <div
+>     class="thumb"
+>     tabindex="0"
+>     role="slider"
+>     aria-valuenow="0"
+>     aria-valuemin="0"
+>     aria-valuemax="10"
+>   />
+>   <input name="{name}" type="hidden" value="0" />
+> </div>
+> ```
+
+## Component anatomy
+
+> Outline the parts of a component; what their names and purposes are. Ideally order them in DOM order. Also consider if the component should be broken into subcomponents. Example:
+>
+> | Name        | Considerations                                                       |
+> | ----------- | -------------------------------------------------------------------- |
+> | `root`      | The root element.                                                    |
+> | `rail`      | The line behind the Slider thumb and rail.                           |
+> | `mark`      | Optional mark line on top of the rail, below the track.              |
+> | `markLabel` | A label aligned with a mark to render the textual value of the mark. |
+> | `track`     | The selected area of the Slider.                                     |
+> | `thumb`     | The focusable draggable knob within the Slider.                      |
+> | `tooltip`   | The tooltip rendered above a thumb.                                  |
+
+## Component props
+
+> Outline the recommended props for the component; taking into consideration conformance guidelines. Make sure to discuss differences and rationalizations.
+>
+> Example:
+>
+> | Name           | Type (default value if clarification needed) | Purpose                                                                                    |
+> | -------------- | -------------------------------------------- | ------------------------------------------------------------------------------------------ |
+> | `as`           | `string` (`"div"`)                           | Replaces the primitive root element type.                                                  |
+> | `className`    | `string`                                     | Allows the author to add a class name to the root element.                                 |
+> | `defaultValue` | `number` \| `number[]`                       | Provides the uncontrolled default value of the component, mutually exclusive with `value`. |
```

Comment:
Open UI specs are framework agnostic so we should avoid React-isms.  That would mean "props" would probably just be "attributes" and the camelCased React-isms would not be present.